### ISSUE TITLE
Low CPU utilization during second phase

### ISSUE THREAD

**Issue by sovaa** [DmitryUlyanov/Multicore-TSNE#9]
It says in "future work" that phase 2 can be improved, so maybe this is what you're referring to?

On phase 2 I see very low CPU utilization (see attached screenshot). Running Multicore-TSNE on 38 cores only utilize 1 core to 100% while the others lie at around 8%.

![mc-tsne-low-cpu](https://cloud.githubusercontent.com/assets/213085/23940505/1997225e-09a0-11e7-890e-5ce49bc3f22d.png)

**Comment by DmitryUlyanov** [comment]
Exactly, phase 2 is not paralellized at all.

**Comment by ShobiStassen** [comment]
i am finding that during stage two my CPUs are used in parallel in a periodic manner: 
![cpu usage multicore tsne](https://user-images.githubusercontent.com/35950152/39913175-85798496-5533-11e8-9a98-c2a0edd6dbda.png)
